[profiler] Standard performance event names for the profiler

digantdesai opened the pull request:
Summary: The goal is to create a hardware/backend independent event abstraction on which a standard set of tooling can be developed.

Test Plan: CI

Reviewed By: kimishpatel

Differential Revision: D40238034

salilsdesai reviewed (approved):
![image](https://user-images.githubusercontent.com/25376640/197847639-41e46d8f-cdd7-4a06-8540-75f453d6050e.png)

digantdesai committed:
[profiler] Standard performance event names for the profiler (#87538)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/87538

The goal is to create a hardware/backend independent event abstraction on which a standard set of tooling can be developed.

Test Plan: CI

Reviewed By: kimishpatel

Differential Revision: D40238034

fbshipit-source-id: 76c25f53cc53457e0caaefbed1f36b38d1725595
